manifests/0000_70_dns-operator_00.crd.yaml

```diff
@@ -42,14 +42,59 @@ spec:
+                  duration for which positive responses should be cached. * negativeTTL
+                  which is a duration for which negative responses should be cached.
+                  If this is not configured, OpenShift will configure positive and
+                  negative caching with a default value that is subject to change.
+                  At the time of writing, the default positiveTTL is 900 seconds and
+                  the default negativeTTL is 30 seconds or as noted in the respective
+                  Corefile for your version of OpenShift.'
+                properties:
+                  negativeTTL:
+                    description: "negativeTTL is optional and specifies the amount
+                      of time that a negative response should be cached. \n If configured,
+                      it must be a value of 1s (1 second) or greater up to a theoretical
+                      maximum of several years. This field expects an unsigned duration
+                      string of decimal numbers, each with optional fraction and a
+                      unit suffix, e.g. \"100s\", \"1m30s\", \"12h30m10s\". Values
+                      that are fractions of a second are rounded down to the nearest
+                      second. If the configured value is less than 1s, the default
+                      value will be used. If not configured, the value will be 0s
+                      and OpenShift will use a default value of 30 seconds unless
+                      noted otherwise in the respective Corefile for your version
+                      of OpenShift. The default value of 30 seconds is subject to
+                      change."
+                    pattern: ^(0|([0-9]+(\.[0-9]+)?(ns|us|µs|μs|ms|s|m|h))+)$
+                    type: string
+                  positiveTTL:
+                    description: "positiveTTL is optional and specifies the amount
+                      of time that a positive response should be cached. \n If configured,
+                      it must be a value of 1s (1 second) or greater up to a theoretical
+                      maximum of several years. This field expects an unsigned duration
+                      string of decimal numbers, each with optional fraction and a
+                      unit suffix, e.g. \"100s\", \"1m30s\", \"12h30m10s\". Values
+                      that are fractions of a second are rounded down to the nearest
+                      second. If the configured value is less than 1s, the default
+                      value will be used. If not configured, the value will be 0s
+                      and OpenShift will use a default value of 900 seconds unless
+                      noted otherwise in the respective Corefile for your version
+                      of OpenShift. The default value of 900 seconds is subject to
+                      change."
+                    pattern: ^(0|([0-9]+(\.[0-9]+)?(ns|us|µs|μs|ms|s|m|h))+)$
```

Comment:
As I mentioned in chat, the pattern shouldn't allow dots if the end result has to be an integer.  The issue with 0.5s or 0.999s being converted to 0s, then replaced by 30s, seems like a bug to me.  I realize this is in the API, and you have an approval already, so I feel bad having to point it out.

Reply:
Does that imply that we wouldn't consider something like `0.5h` as a valid value?

Reply:
Maybe another way to put it...do we want to explicitly disallow fractional values?
Correct: `30m`
Incorrect: `0.5h`

Reply:
The biggest issue is that values between 0.5s and 0.999s end up as 30s.  Maybe if you use https://github.com/openshift/cluster-dns-operator/pull/335#discussion_r984777022 that issue would go away.  

However, a test of an update on a dns configuration that started with 0.5s is still warranted to make sure it doesn't cause a bug

Reply:
I manually tested the following:
`900ms` rounds to `1s` resulting in `1` in the Corefile.
`0.4s` rounds to `0s` resulting in the default of `30` in the Corefile
`1.0004s` rounds to `1s` resulting in `1` in the Corefile.
`900000000ms` rounds to `900000s` resulting in `900000` in the Corefile.

I also set the following values and deleted a CoreDNS pod:
```
spec:
  cache:
    negativeTTL: 0.5s
    positiveTTL: 1.2h33.8m10.1s
```
The pod was able to restart without issue and the following config landed in the Corefile after pod restart:
```
    cache 6358 {
        denial 9984 1
    }
```
Are there any other values you want tested?

Reply:
Oh, you mean after the changes you did this?

Reply:
Correct. I implemented a version of what you suggested in this [comment](https://github.com/openshift/cluster-dns-operator/pull/335#discussion_r984777022). Instead of using `math` I used https://pkg.go.dev/time#Duration.Round